src/ext/framework/help_commands.rs

```diff
@@ -0,0 +1,183 @@
+    let _ = ctx.send_message(message.channel_id, |m| {
+        m.embed(|e| {
+            e.colour(Colour::dark_red())
+                .description(input)
+        })
+    });
+}
+
+#[allow(dead_code)]
+pub fn with_embeds(ctx: &Context,
+                   message: &Message,
+                   groups: HashMap<String, Arc<CommandGroup>>,
+                   args: Vec<String>) {
+    if args.len() > 0 {
+        let name = args.join(" ");
+        for (_, group) in groups {
+            let mut found: Option<(&String, &Command)> = None;
+            if let Some(ref prefix) = group.prefix {
+                for (command_name, command) in &group.commands {
+                    if name == format!("{} {}", prefix, command_name) {
+                        found = Some((command_name, command));
+                    }
+                }
+            } else {
+                for (command_name, command) in &group.commands {
+                    if name == command_name[..] {
+                        found = Some((command_name, command));
+                    }
+                }
+            };
+            if let Some((command_name, command)) = found {
+                if !command.help_available {
+                    error_embed(&ctx, &message, "**Error**: No help available.");
+                    return;
+                }
+                let _ = ctx.send_message(message.channel_id, |m| {
+                    m.embed(|e| {
+                        let mut embed = e.colour(Colour::rosewater())
+                            .title(command_name);
+                        if let Some(ref desc) = command.desc {
+                            embed = embed.field(|f| {
+                                f.name("Description")
+                                    .value(&desc)
+                                    .inline(false)
+                            });
+                        }
+                        if let Some(ref usage) = command.usage {
+                            embed = embed.field(|f| {
+                                f.name("Usage")
+                                    .value(&format!("{} {}", command_name, usage))
+                            });
+                        }
+                        embed = embed.field(|f| {
+                            f.name("Available")
+                                .value(if command.dm_only {
+                                    "Only in DM"
+                                } else if command.guild_only {
+                                    "Only in guilds"
+                                } else {
+                                    "Everywhere"
+                                })
+                        });
+                        embed
+                    })
+                });
+                return;
+            }
+        }
+        error_embed(&ctx,
+                    &message,
+                    &format!("**Error**: Command `{}` not found.", name));
```

Comment:
Same here with ctx and message: already borrowed.

If the resultant value of `format!` is bound to a binding, it should let you do `error_embed(ctx, message, error_msg);`